ArduCopter/mode_guided.cpp

```diff
@@ -491,39 +491,39 @@ bool ModeGuided::set_destination(const Location& dest_loc, bool use_yaw, float y
     // initialise terrain following if needed
     if (is_terrain_alt) {
         // get current alt above terrain
-        float origin_terr_offset_cm;
-        if (!wp_nav->get_terrain_offset_cm(origin_terr_offset_cm)) {
+        float origin_terr_offset_m;
+        if (!wp_nav->get_terrain_offset_m(origin_terr_offset_m)) {
             // if we don't have terrain altitude then stop
             init(true);
             return false;
         }
         // convert origin to alt-above-terrain if necessary
         if (!guided_is_terrain_alt) {
             // new destination is alt-above-terrain, previous destination was alt-above-ekf-origin
-            pos_control->init_pos_terrain_U_cm(origin_terr_offset_cm);
+            pos_control->init_pos_terrain_U_m(origin_terr_offset_m);
         }
     } else {
-        pos_control->init_pos_terrain_U_cm(0.0);
+        pos_control->init_pos_terrain_U_m(0.0);
     }
 
-    guided_pos_target_neu_cm = pos_target_neu_cm.topostype();
+    guided_pos_target_neu_m = pos_target_neu_m.topostype();
     guided_is_terrain_alt = is_terrain_alt;
-    guided_vel_target_neu_cms.zero();
-    guided_accel_target_neu_cmss.zero();
+    guided_vel_target_neu_ms.zero();
+    guided_accel_target_neu_mss.zero();
     update_time_ms = millis();
 
     // log target
 #if HAL_LOGGING_ENABLED
-    copter.Log_Write_Guided_Position_Target(guided_mode, Vector3f(dest_loc.lat, dest_loc.lng, dest_loc.alt), guided_is_terrain_alt, guided_vel_target_neu_cms, guided_accel_target_neu_cmss);
+    copter.Log_Write_Guided_Position_Target(guided_mode, Vector3f(dest_loc.lat, dest_loc.lng, dest_loc.alt), guided_is_terrain_alt, guided_vel_target_neu_ms, guided_accel_target_neu_mss);
```

Comment:
I think the Vector3f(dest_loc.lat, xxx) needs a multiplier because we've changed the argument to accept meters

Reply:
yes you're right.  we've always just logged the lat,lon,alt like this.

Reply:
I thought about this again and this will actually change the units of our logging because previously within Log_Write_Guided_Position_Target() we multiplied by 0.01.  the lat,lon,alt numbers will now be logged as 100x bigger.  Trying to log lat,lon and alt with this number is dumb anyway though, as PeterB says in another comment we should create a new log message.

I've created this issue https://github.com/ArduPilot/ardupilot/issues/30893